code/game/objects/effects/decals/cleanable/humans.dm

```diff
@@ -296,3 +302,96 @@
 	if((blood_state != BLOOD_STATE_OIL) && (blood_state != BLOOD_STATE_NOT_BLOODY))
 		return TRUE
 	return FALSE
+
+/obj/effect/decal/cleanable/blood/hitsplatter
+	name = "blood splatter"
+	pass_flags = PASSTABLE | PASSGRILLE
+	icon_state = "hitsplatter1"
+	random_icon_states = list("hitsplatter1", "hitsplatter2", "hitsplatter3")
+	/// The turf we just came from, so we can back up when we hit a wall
+	var/turf/prev_loc
+	/// Who's the blood donor, so to speak?
+	var/mob/living/blood_source
```

Comment:
Seems like these references need to be handled in `Destroy()`? `blood_source` may be better as a weakref.

Reply:
looked into the blood DNA stuff and just realized it was just a getter for a list var, redid it to just store that list instead